Mathlib/Algebra/Order/Group/Convex.lean

```diff
@@ -0,0 +1,350 @@
+/-
+Copyright (c) 2025 Judith Ludwig and Junyan Xu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Judith Ludwig, Junyan Xu
+-/
+import Mathlib.Algebra.Group.Subgroup.Order
+import Mathlib.Algebra.Order.Archimedean.Class
+import Mathlib.Data.Finite.Card
+import Mathlib.Data.Real.Embedding
+import Mathlib.Data.Set.Card
+import Mathlib.GroupTheory.QuotientGroup.Defs
+import Mathlib.Order.Birkhoff
+
+/-! # Convex subgroups of a linearly ordered abelian group -/
+
+variable {α β : Type*} [CommGroup α] [LinearOrder α] [CommGroup β] [LinearOrder β]
+
+/-- A subgroup `H` of a linearly ordered abelian group is convex, if for any `a ≤ b ≤ 1`,
+`a ∈ H` implies `b ∈ H`. -/
+@[to_additive /-- A subgroup `H` of a linearly ordered additive abelian group is convex,
+if for any `a ≤ b ≤ 0`, `a ∈ H` implies `b ∈ H`. -/]
+def IsConvexSubgroup (H : Subgroup α) : Prop :=
+  ∀ ⦃a b : α⦄, a ≤ b → b ≤ 1 → a ∈ H → b ∈ H
+
+/-- The type of convex subgroups of a linearly ordered additive abelian group. -/
+structure ConvexAddSubgroup (α) [AddCommGroup α] [LinearOrder α] extends AddSubgroup α where
+  convex : IsConvexAddSubgroup toAddSubgroup
+
+variable (α) in
+/-- The type of convex subgroups of a linearly ordered abelian group. -/
+@[to_additive (attr := ext)] structure ConvexSubgroup extends Subgroup α where
+  convex : IsConvexSubgroup toSubgroup
+
+@[to_additive] lemma ConvexSubgroup.toSubgroup_injective :
+    Function.Injective (ConvexSubgroup.toSubgroup : ConvexSubgroup α → Subgroup α) :=
+  fun ⟨_, _⟩ ⟨_, _⟩ ↦ by simp
+
+section
+
+variable [IsOrderedMonoid α]
+
+@[to_additive] lemma IsConvexSubgroup.iff_mem_and_mem_of_mul_mem {H : Subgroup α} :
+    IsConvexSubgroup H ↔ ∀ ⦃a b : α⦄, a ≤ 1 → b ≤ 1 → a * b ∈ H → a ∈ H ∧ b ∈ H where
+  mp convex a _ ale0 ble1 h :=
+    ⟨convex ((mul_le_iff_le_one_right' a).mpr ble1) ale0 h,
+      convex ((mul_le_iff_le_one_left').mpr ale0) ble1 h⟩
+  mpr h a b aleb ble1 ainH :=
+    (h (mul_inv_le_one_iff_le.mpr aleb) ble1 <| (inv_mul_cancel_right a b).symm ▸ ainH).right
+
+@[to_additive] lemma IsConvexSubgroup.iff_ordConnected {H : Subgroup α} :
+    IsConvexSubgroup H ↔ (H : Set α).OrdConnected where
```

Comment:
Why don't we just use `OrdConnected`?

Reply:
I think we won't change the definition of IsConvexSubgroup because it's easier to verify than OrdConnected.